File: ios_app/iphone/TriggersListIconsView.swift
import SwiftUI
import UIKit
import shared

private let emojiSpacing = 6.0

struct TriggersListIconsView: View {

    let triggers: [TextFeatures.Trigger]
    let fontSize: CGFloat
    var leadingPadding: CGFloat = emojiSpacing

    var body: some View {
        if triggers.isEmpty {
            EmptyView()
        } else {
            HStack(spacing: emojiSpacing) {
                ForEach(triggers, id: \.id) { trigger in
                    Button(
                            action: {
                                trigger.performUI()
                            },
                            label: {
                                Text(trigger.emoji)
                                        .font(.system(size: fontSize))
                            }
                    )
                }
            }
                    .padding(.leading, leadingPadding)
        }
    }
}

File: android_app/src/main/java/me/timeto/app/ui/TriggersListIconsView.kt
package me.timeto.app.ui

import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.unit.TextUnit
import androidx.compose.ui.unit.dp
import me.timeto.app.HStack
import me.timeto.shared.TextFeatures

private val triggerShape = RoundedCornerShape(1.dp)

@Composable
fun TriggersListIconsView(
    triggers: List<TextFeatures.Trigger>,
    fontSize: TextUnit,
) {
    if (triggers.isEmpty())
        return

    HStack {
        triggers.forEach { trigger ->
            Text(
                text = trigger.emoji,
                modifier = Modifier
                    .padding(start = 6.dp)
                    .clip(triggerShape)
                    .clickable {
                        trigger.performUI()
                    },
                fontSize = fontSize,
            )
        }
    }
}
